catalog/async/catalog-async-data/src/main/java/org/codice/ddf/catalog/async/data/impl/LazyProcessResourceImpl.java

```diff
@@ -86,7 +89,16 @@ public LazyProcessResourceImpl(String metacardId, Supplier<Resource> resourceSup
   @Override
   public InputStream getInputStream() throws IOException {
     loadResource();
-    return inputStream;
+    if (inputStreamDataCache == null) {
+      if (inputStream == null) {
+        throw new IOException(String.format("Tried to get input stream for %s but was null", name));
+      }
+      inputStreamDataCache = new TemporaryFileBackedOutputStream(32 * 1024);
+      IOUtils.copyLarge(inputStream, inputStreamDataCache);
+      IOUtils.closeQuietly(inputStream);
+    }
+
+    return inputStreamDataCache.asByteSource().openStream();
```

Comment:
❗️ Because the `close()` method also means that the input stream returned will be closed, we should keep track of all returned input streams to automatically close them. This can easily be done with the `com.google.common.io.Closer` class or something like done in `org.codice.solr.factory.impl.Closer` which actually doesn't stop at the first close exception but ignores it and continue on to closing all.